app/Platform/Actions/UpdateGameSetFromGameAlternativesModification.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\Game;
+use App\Models\GameSet;
+use App\Models\System;
+use App\Platform\Enums\GameSetType;
+
+class UpdateGameSetFromGameAlternativesModification
+{
+    public function execute(int $parentGameId, int $childGameId, bool $isAttaching = true): void
+    {
+        $parentGame = Game::find($parentGameId);
+        $childGame = Game::find($childGameId);
+
+        $parentGameSet = GameSet::firstWhere('game_id', $parentGameId);
+        if (!$parentGameSet) {
+            if ($parentGame->ConsoleID === System::Hubs) {
+                $parentGameSet = GameSet::updateOrCreate(
+                    ['game_id' => $parentGame->id],
+                    ['title' => $parentGame->title, 'type' => GameSetType::HUB]
+                );
+            } else {
+                $parentGameSet = GameSet::updateOrCreate(
+                    ['game_id' => $parentGame->id],
+                    ['type' => GameSetType::GAME]
+                );
+            }
+        }
+
+        if ($parentGame->ConsoleID === System::Hubs || $childGame->ConsoleID === System::Hubs) {
```

Comment:
It would probably be simpler to just reverse the relationship than check both. Also, I think this would address an issue where the child is a hub, but the hub doesn't yet exist.
```
if ($childGame->ConsoleID === System::Hubs) {
    if ($parentGame->ConsoleID === System::Hubs) {
        // adding hub to hub, attach link
        return;
    }

    // adding hub to game, swap parent and child so parent is hub
    $temp = $childGame;
    $childGame = $parentGame;
    $parentGame = $temp;
}

// find and create parent game set (per above)
// add game to parent game set
```

Reply:
This took quite some time to wrap my brain around, but I think I have a grasp of what's happening here. Changes should be applied in both places in latest.